src/include/timestamp.i

```diff
@@ -104,46 +104,50 @@
           (ta)->newest_stop_durable_ts == WT_TS_NONE && (ta)->oldest_start_ts == WT_TS_NONE && \
-          (ta)->oldest_start_txn == WT_TXN_NONE && (ta)->newest_stop_ts == WT_TS_MAX &&        \
+          (ta)->newest_txn == WT_TXN_NONE && (ta)->newest_stop_ts == WT_TS_MAX &&              \
           (ta)->newest_stop_txn == WT_TXN_MAX && (ta)->prepare == 0) :                         \
         ((ta)->newest_start_durable_ts == WT_TS_NONE &&                                        \
           (ta)->newest_stop_durable_ts == WT_TS_NONE && (ta)->oldest_start_ts == WT_TS_MAX &&  \
-          (ta)->oldest_start_txn == WT_TXN_MAX && (ta)->newest_stop_ts == WT_TS_NONE &&        \
+          (ta)->newest_txn == WT_TXN_NONE && (ta)->newest_stop_ts == WT_TS_NONE &&             \
           (ta)->newest_stop_txn == WT_TXN_NONE && (ta)->prepare == 0))
 
 /* Copy the values from one time aggregate structure to another. */
 #define WT_TIME_AGGREGATE_COPY(dest, source) (*(dest) = *(source))
 
 /* Update the aggregated window to reflect for a new time window. */
-#define WT_TIME_AGGREGATE_UPDATE(session, ta, tw)                                 \
-    do {                                                                          \
-        WT_ASSERT(session, (ta)->init_merge == 1);                                \
-        (ta)->newest_start_durable_ts =                                           \
-          WT_MAX((tw)->durable_start_ts, (ta)->newest_start_durable_ts);          \
-        (ta)->newest_stop_durable_ts =                                            \
-          WT_MAX((tw)->durable_stop_ts, (ta)->newest_stop_durable_ts);            \
-        (ta)->oldest_start_ts = WT_MIN((tw)->start_ts, (ta)->oldest_start_ts);    \
-        (ta)->oldest_start_txn = WT_MIN((tw)->start_txn, (ta)->oldest_start_txn); \
-        (ta)->newest_stop_ts = WT_MAX((tw)->stop_ts, (ta)->newest_stop_ts);       \
-        (ta)->newest_stop_txn = WT_MAX((tw)->stop_txn, (ta)->newest_stop_txn);    \
-        if ((tw)->prepare != 0)                                                   \
-            (ta)->prepare = 1;                                                    \
+#define WT_TIME_AGGREGATE_UPDATE(session, ta, tw)                              \
+    do {                                                                       \
+        WT_ASSERT(session, (ta)->init_merge == 1);                             \
+        (ta)->newest_start_durable_ts =                                        \
+          WT_MAX((tw)->durable_start_ts, (ta)->newest_start_durable_ts);       \
+        (ta)->newest_stop_durable_ts =                                         \
+          WT_MAX((tw)->durable_stop_ts, (ta)->newest_stop_durable_ts);         \
+        (ta)->oldest_start_ts = WT_MIN((tw)->start_ts, (ta)->oldest_start_ts); \
+        (ta)->newest_txn = WT_MAX((tw)->start_txn, (ta)->newest_txn);          \
```

Comment:
It deserves a comment as to why we double assign to newest_txn

Reply:
Added a comment to explain the reason for double assignment and also added some comments to the aggregated time window structure.